provider/snowflake.go

```diff
@@ -333,3 +344,22 @@ func (sf *snowflakeOfflineStore) UpdateTrainingSet(def TrainingSetDef) error {
 func (sf *snowflakeOfflineStore) AsOfflineStore() (OfflineStore, error) {
 	return sf, nil
 }
+
+// handleErr attempts to add the Snowflake query and session IDs to a wrapped error to aid
+// in further debugging in the Snowflake UI; note that handleErr will not fail even if the
+// returned error isn't an instance of gosnowflake.SnowflakeError or if the query to get
```

Comment:
nit: could unit test the case that nothing explodes for non-SnowflakeError cases